clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -385,15 +389,26 @@ object GarbageCollector {
   ) = {
     val reportLogsDst = concatToGCLogsPrefix(storageNSForHadoopFS, "summary")
     val reportExpiredDst = concatToGCLogsPrefix(storageNSForHadoopFS, "expired_addresses")
+    val addressesMarkMetadataLocation = getMetadataMarkLocation(markId, gcAddressesLocation)
+    val addressesMarkMetadataDF = spark.read.json(addressesMarkMetadataLocation)
     val time = DateTimeFormatter.ISO_INSTANT.format(java.time.Clock.systemUTC.instant())
     writeParquetReport(commitsDF, reportLogsDst, time, "commits.parquet")
     writeParquetReport(expiredAddresses, reportExpiredDst, time)
     writeJsonSummary(configMapper, reportLogsDst, removed.count(), gcRules, time)
 
+    import spark.implicits._
+    val metadataArray = addressesMarkMetadataDF
+      .select(RUN_ID_KEY)
+      .map(_.getString(0))
+      .collect
+
+    val runID = metadataArray(0)
```

Comment:
Can you please help me understand the following: 
- I'm not sure I understand the relationship between mark and run ids: is it one-to-many in the sense that a markId can be associated with multiple run ids? 
- Why would this runID be different than the id in line 294? 
Thanks!



Why would the runId associated with a markId be

Reply:
* Since a mark ID can be provided by the user, multiple run IDs might have the same mark ID (1-n relationship). This is possible if the user wants to overwrite the previous mark ID.
* The run ID in line 294 is generated if a "mark" had occurred (it's generated in the prepareCommits API call in lakeFS). If this is a "sweep-only" run, then we should fetch the run ID from the metadata of the associated mark ID. Having said that, the next PR (spoiler alert) will pass the run ID to `writeReports`